### PR TITLE
feat: Support remotely-managed cloudflare tunnels

### PR DESCRIPTION
**Pull request by jfroy**

### CODE REVIEW

File: bootstrap/templates/kubernetes/apps/network/cloudflared/app/secret.sops.yaml.j2 [hunk 1]
```diff
@@ -5,6 +5,7 @@ metadata:
   name: cloudflared-secret
 stringData:
   TUNNEL_ID: "#{ bootstrap_cloudflare.tunnel.id }#"
+  TUNNEL_TOKEN: "#{ bootstrap_cloudflare.tunnel.token }#"
```
Comment (onedr0p) [inline bootstrap/templates/kubernetes/apps/network/cloudflared/app/secret.sops.yaml.j2:8]: This can be wrapped in an if statement

**Comment by onedr0p** [comment]
I wonder if there's a better way to name locally vs remotely managed tunnels? It might be because it's early but this confused me a bit at first 😅

File: bootstrap/templates/kubernetes/apps/network/cloudflared/app/secret.sops.yaml.j2 (hunk 1 above)
Comment (onedr0p) [inline bootstrap/templates/kubernetes/apps/network/cloudflared/app/secret.sops.yaml.j2:8]:
```suggestion
  #% if bootstrap_cloudflare.tunnel.token %#
  TUNNEL_TOKEN: "#{ bootstrap_cloudflare.tunnel.token }#"
  #% endif %#
```

**Comment by jfroy** [comment]
> I wonder if there's a better way to name locally vs remotely managed tunnels? It might be because it's early but this confused me a bit at first 😅
> 
> 
> 
> Maybe cli-created vs web-created?

I borrowed those terms straight from cloudflare for consistency. https://developers.cloudflare.com/cloudflare-one/connections/connect-networks/get-started/

**Comment by mrueg** [comment]
Alternatively, we could use https://github.com/adyanth/cloudflare-operator which would allow changes after the cluster has been bootstrapped.

**Comment by jfroy** [comment]
> Alternatively, we could use https://github.com/adyanth/cloudflare-operator which would allow changes after the cluster has been bootstrapped.

Technically the only thing that's required to use a remotely-managed tunnel is to set the `TUNNEL_TOKEN` env var. If it is set to a non-empty value, `cloudflared` will use it, otherwise it will continue and use the credentials file. So all someone would need to do is edit the secret. We would need to edit the PR to not have the template conditionals in the secret definition and in the helm release.

https://github.com/cloudflare/cloudflared/blob/master/cmd/cloudflared/tunnel/subcommands.go#L656

**Comment by onedr0p** [comment]
> Alternatively, we could use https://github.com/adyanth/cloudflare-operator which would allow changes after the cluster has been bootstrapped.

That would be nice but the release and commit history on it doesn't give me confidence that it's well supported.

File: bootstrap/templates/kubernetes/apps/network/cloudflared/app/secret.sops.yaml.j2 (hunk 1 above)
Comment (onedr0p) [inline bootstrap/templates/kubernetes/apps/network/cloudflared/app/secret.sops.yaml.j2:8]: Does this render out fine and encrypt with sops if the value is not set?

Comment (jfroy) [reply bootstrap/templates/kubernetes/apps/network/cloudflared/app/secret.sops.yaml.j2:8]: Sorry for the delay, was away. I verified this indeed renders and encrypts fine.

**Comment by onedr0p** [comment]
@jfroy I am getting some support requests that make this a bit confusing to people. I may just go back to CLI-only created and managed tunnels.

**Comment by jfroy** [comment]
> @jfroy I am getting some support requests that make this a bit confusing to people. I may just go back to CLI-only created and managed tunnels.

Totally fair. I wasn't comfortable with yet more secrets in plaintext under my home directory, so I went with a dashboard tunnel.

**Pull request merged**